satpy/readers/seviri_l1b_native.py

```diff
@@ -443,13 +443,7 @@ def get_dataset(self, dataset_id, dataset_info):
             data0 = dec10216(raw0.flatten())
             data0 = data0.reshape(shape_layer)
 
-            data = np.zeros(shape)
-            idx = range(0, shape[0], 3)
-            data[idx, :] = data0
-            idx = range(1, shape[0], 3)
-            data[idx, :] = data1
-            idx = range(2, shape[0], 3)
-            data[idx, :] = data2
+            data = da.stack((data0, data1, data2), axis=1).reshape(shape)
```

Comment:
Can you use `np.stack` instead while still preserving the array as dask arrays?

Reply:
Also, I think using the dedicated method will make it more readable, ie using
`hstack(arrays)` instead of `stack(arrays, axis=1)`.

Reply:
Thanks for your review! I added a commit changing from `da.stack` to `np.stack`. 
I tried using `np.hstack`, but I get following error:
```
  File "/tcenas/home/andream/code/satpy_latest/satpy/satpy/readers/seviri_l1b_native.py", line 446, in get_dataset
    data = np.hstack((data0, data1, data2)).reshape(shape)
  File "/tcenas/home/andream/anaconda3/envs/trolldev_latest/lib/python3.7/site-packages/dask/array/core.py", line 1758, in reshape
    return reshape(self, shape)
  File "/tcenas/home/andream/anaconda3/envs/trolldev_latest/lib/python3.7/site-packages/dask/array/reshape.py", line 195, in reshape
    inchunks, outchunks = reshape_rechunk(x.shape, shape, x.chunks)
  File "/tcenas/home/andream/anaconda3/envs/trolldev_latest/lib/python3.7/site-packages/dask/array/reshape.py", line 64, in reshape_rechunk
    raise ValueError("Shapes not compatible")
ValueError: Shapes not compatible
```
the error disappears if I change `data0, data1, data2` to be numpy arrays, so it appears that it is something specific to the dask reshaping/rechunking. I couldn't find a solution to get past this.

Reply:
ok, good to know! thanks for looking into this.